src/core/thread/mesh_forwarder.cpp

```diff
@@ -888,7 +901,10 @@ void MeshForwarder::SetPollPeriod(uint32_t aPeriod)
             mPollPeriod = aPeriod;
         }
 
-        mPollTimer.Start(mPollPeriod);
+        if (mPollTimer.IsRunning() && ((mNetif.GetMle().GetDeviceMode() & Mle::ModeTlv::kModeFFD) == 0))
```

Comment:
`SetRxOnWhenIdle()` is responsible for starting `mPollTimer`.  This change makes sure that only `SetRxOnWhenIdle()` can start `mPollTimer`.